process/indexer/indexer.go

```diff
@@ -0,0 +1,37 @@
+package indexer
+
+import (
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-sovereign-notifier-go/process"
+)
+
+type indexer struct {
+	notifier process.SovereignNotifier
+}
+
+// NewIndexer creates an indexer which will internally save received blocks
+// and notify sovereign shards for each finalized block
+func NewIndexer(notifier process.SovereignNotifier) (process.Indexer, error) {
+	if notifier == nil {
+		return nil, errNilSovereignNotifier
+	}
+
+	return &indexer{
+		notifier: notifier,
+	}, nil
+}
+
+// SaveBlock will save the received block in an internal cache
```

Comment:
Update the comments or add TODOs for future implementation

Reply:
These are updated, those implementations will follow in the immediate next PR